R/generateHypotheses.R

```diff
@@ -1,192 +1,305 @@
-#' generateBinaryNetwork
+#' 
+#' Generate a binary network from complexes.
 #' @description Generate binary network from concrete complex hypotheses (e.g. CORUM).
 #' @import data.table
-#' @param complex_hypothesis data.table in the format of complex hypotheses
+#' @param complex_hypotheses data.table specifying complex hypotheses.
+#' Must have the following columns:
+#' \itemize{
+#' \item complex_id: character, a unique id for every complex
+#' \item protein_id: character, the protein id, e.g. Uniprot id
+#' }
 #' @return data.table with binary interactions between "a" and "b"
 #' @export
-generateBinaryNetwork <- function(complex_hypothesis){
-  complexes <- unique(complex_hypothesis$complex_id)
-  binary_interactions <- data.table(a=character(0),b=character(0))
-  # get all binary interactions for each complex in the complex hypothesis
-  for(i in c(1:length(complexes))){
-    binary_interactions <- appendBinaryInteractions(complex=complexes[i],hypothesis=complex_hypothesis,binary_interactions=binary_interactions)
-  }
-  setkey(binary_interactions,NULL)
-  unique(binary_interactions) # remove duplicates
+#' @examples 
+#' ## Load example Data
+#' complexHypotheses <- exampleComplexHypotheses
+#' 
+#' ## Generate the binary network
+#' binaryInteractions <- generateBinaryNetwork(complex_hypotheses = complexHypotheses)
+#' ##Inspect the result
+#' binaryInteractions
+#' 
+#' 
+generateBinaryNetwork <- function(complex_hypotheses){
+  setkey(complex_hypotheses, "complex_id")
+  complexes <- unique(complex_hypotheses$complex_id)
+  ## Get every combination of binary interactions within a complex
+  binary_interactions <- do.call("rbind", lapply(complexes, function(x){
+    t(combn(complex_hypotheses[complex_id == x, protein_id], m = 2))
+  }))
+  ## Keep only unique interactions
```

Comment:
I guess that sorting and setting a unique subset is faster if you first convert to data.table.
What do you think?